clearnode/pkg/rpc/client_manual_test.go

```diff
@@ -0,0 +1,305 @@
+	"github.com/erc7824/nitrolite/clearnode/pkg/sign"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/signer/core/apitypes"
+	"github.com/google/uuid"
+	"github.com/shopspring/decimal"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/net/context"
+)
+
+const (
+	sandboxWsRpcUrl = "wss://clearnet-sandbox.yellow.com/ws"
+)
+
+func TestManualClient(t *testing.T) {
+	walletPK := os.Getenv("TEST_WALLET_PK")
+	if walletPK == "" {
+		t.Skip("TEST_WALLET_PK not set, skipping manual client test")
+	}
+	walletSigner, err := sign.NewEthereumSigner(walletPK)
+	require.NoError(t, err)
+	fmt.Printf("Using wallet address: %s\n", walletSigner.PublicKey().Address().String())
+
+	sessionPK := os.Getenv("TEST_SESSION_PK")
+	if sessionPK == "" {
+		t.Skip("TEST_SESSION_PK not set, skipping manual client test")
+	}
+	sessionSigner, err := sign.NewEthereumSigner(sessionPK)
+	require.NoError(t, err)
+	fmt.Printf("Using session address: %s\n", sessionSigner.PublicKey().Address().String())
+
+	dialer := rpc.NewWebsocketDialer(rpc.DefaultWebsocketDialerConfig)
+	client := rpc.NewClient(dialer)
+
+	ctx, cancel := context.WithCancel(t.Context())
+
+	errCh := make(chan error, 1)
+	handleError := func(err error) {
+		errCh <- err
+	}
+
+	go dialer.Dial(ctx, sandboxWsRpcUrl, handleError)
+	go client.ListenEvents(ctx, handleError)
+
+	for !dialer.IsConnected() {
+		select {
+		case err := <-errCh:
+			if err != nil {
+				t.Fatalf("connection error: %v", err)
+			}
+		default:
+			time.Sleep(100 * time.Millisecond)
+		}
+	}
+
+	var jwtToken string
+	t.Run("Authenticate With Signature", func(t *testing.T) {
+		authReq := rpc.AuthRequestRequest{
+			Address:            walletSigner.PublicKey().Address().String(),
+			SessionKey:         sessionSigner.PublicKey().Address().String(),
+			AppName:            "TestClient",
+			Allowances:         []rpc.Allowance{},
+			Expire:             "",
+			Scope:              "",
+			ApplicationAddress: walletSigner.PublicKey().Address().String(),
+		}
+		authRes, _, err := client.AuthRequest(ctx, authReq)
+		require.NoError(t, err)
+
+		challengeSig, err := signChallenge(walletSigner, authReq, authRes.ChallengeMessage.String())
+		require.NoError(t, err)
+
+		verifySigReq := rpc.AuthSigVerifyRequest{
+			Challenge: authRes.ChallengeMessage,
+		}
+		verifySigRes, _, err := client.AuthSigVerify(ctx, verifySigReq, challengeSig)
+		require.NoError(t, err)
+		require.True(t, verifySigRes.Success, "auth_sig_verify should succeed")
+		require.NotEmpty(t, verifySigRes.JwtToken, "jwt token should be set")
+		jwtToken = verifySigRes.JwtToken
+	})
+
+	cancel()
+	ctx, cancel = context.WithCancel(t.Context())
+	defer cancel()
+
+	go dialer.Dial(ctx, sandboxWsRpcUrl, handleError)
+	go client.ListenEvents(ctx, handleError)
+
+	for !dialer.IsConnected() {
+		select {
+		case err := <-errCh:
+			if err != nil {
+				t.Fatalf("connection error: %v", err)
+			}
+		default:
+			time.Sleep(100 * time.Millisecond)
+		}
+	}
```

Comment:
This smells really unidiomatic... We should look into it at some point and try redesign the flow